{
  "uid" : "5a1d94e723ac9bb3",
  "name" : "test_valid_email",
  "fullName" : "tests.test_forgot_password.TestForgotPassword#test_valid_email",
  "historyId" : "ad76d87e5a831f6b00837c8ab7fbb24e",
  "time" : {
    "start" : 1604104745460,
    "stop" : 1604104748968,
    "duration" : 3508
  },
  "description" : "Forgot password feature test with a valid email address",
  "descriptionHtml" : "<p>Forgot password feature test with a valid email address</p>\n",
  "status" : "broken",
  "statusMessage" : "TypeError: selenium.webdriver.remote.webdriver.WebDriver.find_element() argument after * must be an iterable, not WebElement",
  "statusTrace" : "self = <tests.test_forgot_password.TestForgotPassword object at 0x7f5793259640>\ncreate_driver = <selenium.webdriver.remote.webdriver.WebDriver (session=\"14aeb1d93ddfdaedf69bc83b7969bbf2\")>\nprep_properties = <utils.config_parser.ConfigParserIni object at 0x7f5793236190>\n\n    @allure.description(\"Forgot password feature test with a valid email address\")\n    def test_valid_email(self, create_driver, prep_properties):\n        config_reader = prep_properties\n        email = config_reader.config_section_dict(\"Base Url\")[\"username\"]\n        ap = AboutPage()\n        lp = LoginPage()\n        fp = ForgotPasswordPage()\n        ap.click_login_link()\n        lp.click_forgot_password()\n        fp.send_password_reset_link(email)\n>       assert_that(fp.get_success_msg()).is_equal_to(self._success_msg)\n\ntests/test_forgot_password.py:28: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/forgot_password_page.py:27: in get_success_msg\n    return self.get_text(self._driver.find_element(*self.SUCCESS_MSG))\npages/base_page.py:40: in get_text\n    el = self._wait.until(expected_conditions.visibility_of_element_located(webelement))\n/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/selenium/webdriver/support/wait.py:71: in until\n    value = method(self._driver)\n/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/selenium/webdriver/support/expected_conditions.py:128: in __call__\n    return _element_if_visible(_find_element(driver, self.locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ndriver = <selenium.webdriver.remote.webdriver.WebDriver (session=\"14aeb1d93ddfdaedf69bc83b7969bbf2\")>\nby = <selenium.webdriver.remote.webelement.WebElement (session=\"14aeb1d93ddfdaedf69bc83b7969bbf2\", element=\"1521e415-a72a-485c-b381-d81e3f3567e3\")>\n\n    def _find_element(driver, by):\n        \"\"\"Looks up an element. Logs and re-raises ``WebDriverException``\n        if thrown.\"\"\"\n        try:\n>           return driver.find_element(*by)\nE           TypeError: selenium.webdriver.remote.webdriver.WebDriver.find_element() argument after * must be an iterable, not WebElement\n\n/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/selenium/webdriver/support/expected_conditions.py:411: TypeError",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "prep_properties",
    "time" : {
      "start" : 1604104650312,
      "stop" : 1604104650312,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "create_driver",
    "time" : {
      "start" : 1604104741819,
      "stop" : 1604104744971,
      "duration" : 3152
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "_session_faker",
    "time" : {
      "start" : 1604104650223,
      "stop" : 1604104650312,
      "duration" : 89
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "testStage" : {
    "description" : "Forgot password feature test with a valid email address",
    "status" : "broken",
    "statusMessage" : "TypeError: selenium.webdriver.remote.webdriver.WebDriver.find_element() argument after * must be an iterable, not WebElement",
    "statusTrace" : "self = <tests.test_forgot_password.TestForgotPassword object at 0x7f5793259640>\ncreate_driver = <selenium.webdriver.remote.webdriver.WebDriver (session=\"14aeb1d93ddfdaedf69bc83b7969bbf2\")>\nprep_properties = <utils.config_parser.ConfigParserIni object at 0x7f5793236190>\n\n    @allure.description(\"Forgot password feature test with a valid email address\")\n    def test_valid_email(self, create_driver, prep_properties):\n        config_reader = prep_properties\n        email = config_reader.config_section_dict(\"Base Url\")[\"username\"]\n        ap = AboutPage()\n        lp = LoginPage()\n        fp = ForgotPasswordPage()\n        ap.click_login_link()\n        lp.click_forgot_password()\n        fp.send_password_reset_link(email)\n>       assert_that(fp.get_success_msg()).is_equal_to(self._success_msg)\n\ntests/test_forgot_password.py:28: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/forgot_password_page.py:27: in get_success_msg\n    return self.get_text(self._driver.find_element(*self.SUCCESS_MSG))\npages/base_page.py:40: in get_text\n    el = self._wait.until(expected_conditions.visibility_of_element_located(webelement))\n/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/selenium/webdriver/support/wait.py:71: in until\n    value = method(self._driver)\n/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/selenium/webdriver/support/expected_conditions.py:128: in __call__\n    return _element_if_visible(_find_element(driver, self.locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ndriver = <selenium.webdriver.remote.webdriver.WebDriver (session=\"14aeb1d93ddfdaedf69bc83b7969bbf2\")>\nby = <selenium.webdriver.remote.webelement.WebElement (session=\"14aeb1d93ddfdaedf69bc83b7969bbf2\", element=\"1521e415-a72a-485c-b381-d81e3f3567e3\")>\n\n    def _find_element(driver, by):\n        \"\"\"Looks up an element. Logs and re-raises ``WebDriverException``\n        if thrown.\"\"\"\n        try:\n>           return driver.find_element(*by)\nE           TypeError: selenium.webdriver.remote.webdriver.WebDriver.find_element() argument after * must be an iterable, not WebElement\n\n/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/selenium/webdriver/support/expected_conditions.py:411: TypeError",
    "steps" : [ {
      "name" : "Click Login link",
      "time" : {
        "start" : 1604104745461,
        "stop" : 1604104746612,
        "duration" : 1151
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    }, {
      "name" : "Click Forgot Password link",
      "time" : {
        "start" : 1604104746612,
        "stop" : 1604104747174,
        "duration" : 562
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    }, {
      "name" : "Send password reset link to email address: 'eshorrosh@gmail.com'",
      "time" : {
        "start" : 1604104747174,
        "stop" : 1604104748901,
        "duration" : 1727
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ {
        "name" : "email",
        "value" : "'eshorrosh@gmail.com'"
      } ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : true
    }, {
      "name" : "Get success message",
      "time" : {
        "start" : 1604104748901,
        "stop" : 1604104748968,
        "duration" : 67
      },
      "status" : "broken",
      "statusMessage" : "TypeError: selenium.webdriver.remote.webdriver.WebDriver.find_element() argument after * must be an iterable, not WebElement\n",
      "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/allure_commons/_allure.py\", line 157, in impl\n    return func(*a, **kw)\n  File \"/home/runner/work/selenium-python/selenium-python/pages/forgot_password_page.py\", line 27, in get_success_msg\n    return self.get_text(self._driver.find_element(*self.SUCCESS_MSG))\n  File \"/home/runner/work/selenium-python/selenium-python/pages/base_page.py\", line 40, in get_text\n    el = self._wait.until(expected_conditions.visibility_of_element_located(webelement))\n  File \"/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/selenium/webdriver/support/wait.py\", line 71, in until\n    value = method(self._driver)\n  File \"/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/selenium/webdriver/support/expected_conditions.py\", line 128, in __call__\n    return _element_if_visible(_find_element(driver, self.locator))\n  File \"/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/selenium/webdriver/support/expected_conditions.py\", line 411, in _find_element\n    return driver.find_element(*by)\n",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : true,
      "hasContent" : true
    } ],
    "attachments" : [ {
      "uid" : "4972aeedbc20452b",
      "name" : "screenshot on failure: 31/10/2020, 00:39:04",
      "source" : "4972aeedbc20452b.png",
      "type" : "image/png",
      "size" : 519743
    }, {
      "uid" : "b2645ec4c7a61de",
      "name" : "screenshot on failure: 31/10/2020, 00:39:08",
      "source" : "b2645ec4c7a61de.png",
      "type" : "image/png",
      "size" : 44508
    } ],
    "parameters" : [ ],
    "stepsCount" : 4,
    "attachmentsCount" : 2,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  },
  "afterStages" : [ {
    "name" : "create_driver::0",
    "time" : {
      "start" : 1604104749284,
      "stop" : 1604104749339,
      "duration" : 55
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "labels" : [ {
    "name" : "severity",
    "value" : "critical"
  }, {
    "name" : "story",
    "value" : "Forgot Password Feature's Functionality"
  }, {
    "name" : "epic",
    "value" : "Security"
  }, {
    "name" : "tag",
    "value" : "security"
  }, {
    "name" : "parentSuite",
    "value" : "tests"
  }, {
    "name" : "suite",
    "value" : "test_forgot_password"
  }, {
    "name" : "subSuite",
    "value" : "TestForgotPassword"
  }, {
    "name" : "host",
    "value" : "fv-az68-995"
  }, {
    "name" : "thread",
    "value" : "4814-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.test_forgot_password"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "critical",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Test defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "tags" : [ "security" ]
  },
  "source" : "5a1d94e723ac9bb3.json",
  "parameterValues" : [ ]
}